extensions/arc/src/ui/tree/resourceTreeNode.ts

```diff
@@ -11,7 +11,7 @@ import { TreeNode } from './treeNode';
  * A TreeNode belonging to a child of a Controller
  */
 export abstract class ResourceTreeNode extends TreeNode {
```

Comment:
Suggestion: make ResourceTreeNode take a type parameter M that is of type ResourceModel or its descendants. Then in the subclasses M can be set to specific Model types avoiding any casts.

Reply:
Nice, I like it